src/ngResource/resource.js

```diff
@@ -658,7 +658,9 @@ angular.module('ngResource', ['ng']).
               httpConfig.timeout = timeoutDeferred.promise;
 
               if (numericTimeout) {
-                numericTimeoutPromise = $timeout(timeoutDeferred.resolve, numericTimeout);
+                numericTimeoutPromise = $timeout(function() {
+                    timeoutDeferred.resolve("timeout");
+                }, numericTimeout);
```

Comment:
This can be also written as `numericTimeoutPromise = $timeout(timeoutDeferred.resolve, numericTimeout, false, 'timeout');`